src/tls/test-server.c

```diff
@@ -375,6 +375,7 @@ teardown (TestCase *tc, gconstpointer data)
   g_assert_cmpint (wait (NULL), ==, -1);
   g_assert_cmpint (errno, ==, ECHILD);
   /* connection should fail */
+  /* coverity[leaked_handle : FALSE] */
   g_assert_cmpint (do_connect (tc), ==, -ECONNREFUSED);
```

Comment:
Annoying that it still doesn't understand this pattern :(

Reply:
I'm not really sure how to rewrite that in a different way. Just storing the result in a variable and then asserting it doesn't fundamentally change the data flow, I suppose it just doesn't know about the intention of test-* code not making sense with assertions being compiled out?

Reply:
Yes, it is annoying.  I tried to extend Coverity's model of `connect()` to set `errno` in the failing case but it did not make this false positive go away.

Reply:
We already model `g_assertion_message()` as `__coverity_panic__()`.  The problem here seems to be that Coverity fails to infer that `fd` is closed if and only if `do_connect()` returns a negative value, which `-ECONNREFUSED` is for sure.